src/utils/daos_hdlr.c

```diff
@@ -1607,25 +1607,26 @@ parse_filename_dfs(const char *path, char **_obj_name, char **_cont_name)
 	char	*f2 = NULL;
 	char	*fname = NULL;
 	char	*cont_name = NULL;
+	int	path_len = strlen(path) + 1;
```

Comment:
We should be avoiding the use of strlen() wherever possible, in this case path seems to come from a stack variable in fs_copy() which is a known size so there's no reason not to pass in the size, however there is another issue here which is that MAX_FILENAME(256) is nowhere near long enough.

char filename[MAX_FILENAME] should be replaced by D_ASPRINTF(), and tracking of len.

Reply:
@ashleypittman this would require changing every single dfs wrapper since parse_filename_dfs is used in just about every single one. I do think improving this function can be looked at under https://jira.hpdd.intel.com/browse/DAOS-6449 since we are refactoring the dfs wrappers under that ticket, and moving them to another location.

Reply:
@ashleypittman I updated the filenames used in fs_copy to use D_ASPRINTF since it is recursively walking the file tree, and you are right 256 was not long enough.